File: app/src/main/java/ru/iandreyshev/parserrss/models/imageProps/FeedListIconProps.java
package ru.iandreyshev.parserrss.models.imageProps;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.support.annotation.NonNull;
import android.util.Log;

import java.io.ByteArrayOutputStream;

public final class FeedListIconProps implements IImageProps {
    private static final String TAG = FeedListIconProps.class.getName();
    private static final Bitmap.CompressFormat FORMAT = Bitmap.CompressFormat.JPEG;
    private static final int QUALITY = 25;
    private static final float SIZE_FACTOR = 0.4f;

    @NonNull
    public static FeedListIconProps newInstance() {
        return new FeedListIconProps();
    }

    @NonNull
    @Override
    public Bitmap configure(@NonNull Bitmap originImage) {
        final int newWidth = (int) (originImage.getWidth() * SIZE_FACTOR);
        final int newHeight = (int) (originImage.getHeight() * SIZE_FACTOR);
        final Bitmap copy = Bitmap.createScaledBitmap(originImage, newWidth, newHeight, false);

        try (ByteArrayOutputStream stream = new ByteArrayOutputStream()) {
            copy.compress(FORMAT, QUALITY, stream);
            byte[] bytes = stream.toByteArray();

            return BitmapFactory.decodeByteArray(bytes, 0, bytes.length);

        } catch (Exception ex) {
            Log.e(TAG, Log.getStackTraceString(ex));
        }

        return originImage;
    }
}

File: app/src/main/kotlin/ru/iandreyshev/parserrss/models/imageProps/FeedListIconProps.kt
package ru.iandreyshev.parserrss.models.imageProps

import ru.iandreyshev.parserrss.models.extention.bitmap
import ru.iandreyshev.parserrss.models.extention.bytes
import ru.iandreyshev.parserrss.models.extention.scaleToSize

class FeedListIconProps : IImageProperties {
    companion object {
        private const val MAX_SIZE = 144
    }

    override fun configureToView(originImage: ByteArray): ByteArray {
        return originImage.bitmap
                .scaleToSize(MAX_SIZE)
                .bytes
    }

    override fun configureToMemory(originImage: ByteArray): ByteArray {
        return originImage.bitmap
                .scaleToSize(IImageProperties.MAX_IMAGE_SIZE)
                .bytes
    }
}
